Cineaste/CoreDataHelper/FetchedResultsManager.swift

```diff
@@ -28,36 +47,170 @@ final class FetchedResultsManager: NSObject {
+            decoder.userInfo[.context] = storageManager.backgroundContext
+
+            do {
+                let movies = try decoder.decode([StoredMovie].self, from: data)
+
+                let dispatchGroup = DispatchGroup()
+
+                //load all posters
+                for movie in movies {
+                    dispatchGroup.enter()
+
+                    movie.loadPoster { poster in
+                        movie.poster = poster
+
+                        dispatchGroup.leave()
+                    }
+                }
+
+                dispatchGroup.notify(queue: .global()) {
+                    storageManager.insertMoviesFromImport(with: movies) { result in
+                        switch result {
+                        case .error:
+                            completionHandler(Result.error(FileImportError.savingNewMovies))
+                        case .success:
+                            completionHandler(Result.success(movies.count))
+                        }
+                    }
+                }
+            } catch {
+                completionHandler(Result.error(FileImportError.parsingJsonToStoredMovie))
+            }
+        }
+    }
+
+    private func saveToDocumentsDirectory(_ data: Data, completionHandler: (Result<Bool>) -> Void) {
+        guard let documentsDirectoryPathString = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true).first,
+            let documentsDirectoryPath = URL(string: "file://\(documentsDirectoryPathString)")
+            else {
+                completionHandler(Result.error(FileExportError.creatingDocumentPath))
+                return
+        }
+
+        let jsonFilePath = documentsDirectoryPath.appendingPathComponent(Strings.exportMoviesFileName)
+        exportMoviesPath = jsonFilePath
+
+        let fileManager = FileManager.default
+        var isDirectory: ObjCBool = false
+
+        // creating a .json file in the Documents folder
+        if !fileManager.fileExists(atPath: jsonFilePath.path, isDirectory: &isDirectory) {
+            let created = fileManager.createFile(atPath: jsonFilePath.path, contents: nil, attributes: nil)
+            guard created == true else {
+                completionHandler(Result.error(FileExportError.creatingFileAtPath))
+                return
+            }
+        }
+
+        // Write that JSON to the file created earlier
+        do {
+            let file = try FileHandle(forWritingTo: jsonFilePath)
+            file.truncateFile(atOffset: 0)
+            file.write(data)
+            completionHandler(Result.success(true))
+        } catch {
+            completionHandler(Result.error(FileExportError.writingContentInFile))
+        }
+    }
+
+    private func resetAllMovies(with storageManager: MovieStorage,
+                                handler: @escaping () -> Void) {
+        let dispatchGroup = DispatchGroup()
+
+        if let storedMovies = controller?.fetchedObjects {
```

Comment:
It seems a bit weird to use the controller for basic data tasks as I use it normally only in the context of a tableview.